library/core/src/ptr/non_null.rs

```diff
@@ -1821,12 +1857,76 @@ mod verify {
         }
     }
 
-    // pub const unsafe fn new(ptr: *mut T) -> Option<Self>
+    // pub const fn new(ptr: *mut T) -> Option<Self>
     #[kani::proof_for_contract(NonNull::new)]
     pub fn non_null_check_new() {
         let mut x: i32 = kani::any();
         let xptr = &mut x;
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    // pub const unsafe fn add(self, count: usize) -> Self
+    #[kani::proof_for_contract(NonNull::add)]
+    pub fn non_null_check_add() {
+        const SIZE: usize = 100000;
+        // Randomize pointer offset within array bound
+        let offset = kani::any_where(|x| *x <= SIZE as isize && *x >=0);
+        //kani::assume(offset >= 0);
+        // Create a non-deterministic array of size SIZE
+        let arr: [i8; SIZE] = kani::any();  
+        // Get a raw pointer to the array
+        let raw_ptr: *mut i8 = arr.as_ptr() as *mut i8;  
+        // NonNull pointer to the random offset
+        let ptr = unsafe { NonNull::new(raw_ptr.offset(offset)).unwrap()};
+        // Create a non-deterministic count value
+        let count: usize = kani::any();  
+        
+        unsafe {
+            // Add a positive offset to pointer
+            let result = ptr.add(count);
+        }
+    }
+
+    // pub fn addr(self) -> NonZero<usize>
+    #[kani::proof_for_contract(NonNull::addr)]
+    pub fn non_null_check_addr() {  
+        // Create NonNull pointer & get pointer address
+        let x = kani::any::<usize>() as *mut i32;
+        kani::assume(!x.is_null());
+        let nonnull_xptr = NonNull::new(x).unwrap();
+        let address = nonnull_xptr.addr();
+    }
+
+    // pub fn align_offset(self, align: usize) -> usize
+    #[kani::proof_for_contract(NonNull::align_offset)]
+    pub fn non_null_check_align_offset() {
+        // Create NonNull pointer
+        let x = kani::any::<usize>() as *mut i32;
+        kani::assume(!x.is_null());
```

Comment:
Instead of using `kani::assume()`, I would prefer to either implement Arbitrary or using:
```rust
let Some(nonnull) = NonNull::new(x) else { return; };
```